src/sentry/incidents/logic.py

```diff
@@ -553,6 +563,26 @@ def create_alert_rule(
 
     resolution = get_alert_resolution(time_window, organization)
 
+    if detection_type == AlertRuleDetectionType.DYNAMIC:
+        if not (sensitivity and seasonality):
+            raise ValidationError("Dynamic alerts require both sensitivity and seasonality")
+    else:
+        if sensitivity or seasonality:
+            raise ValidationError(
+                "Sensitivity and seasonality are not valid fields for this alert type"
+            )
+        if threshold_type == AlertRuleThresholdType.ABOVE_AND_BELOW:
+            raise ValidationError(
+                "Above and below is not a valid threshold type for this alert type"
+            )
+
+    if detection_type == AlertRuleDetectionType.PERCENT:
+        if comparison_delta is None:
+            raise ValidationError("Percentage-based alerts require a comparison delta")
+    else:
+        if comparison_delta is not None:
+            raise ValidationError("Comparison delta is not a valid field for this alert type")
+
     if comparison_delta is not None:
```

Comment:
I just remembered that since we've published these endpoints we need to be careful to not break customer's scripts/workflow/etc. and should make sure that if they're creating a percent type alert but not passing `detection_type` (because it's new and they didn't know to pass it) we set the detection type to percent for them.